front/app/containers/App/index.tsx

```diff
@@ -274,33 +274,40 @@ class App extends PureComponent<Props, State> {
         this.closePostPageModal();
       }),
 
-      eventEmitter.observeEvent('tryAndDeleteProfile').subscribe(() => {
-        signOutAndDeleteAccountPart2().then((success) => {
-          if (success) {
-            this.setState({
-              userDeletedModalOpened: true,
-              userActuallyDeleted: true,
-            });
-          } else {
-            this.setState({
-              userDeletedModalOpened: true,
-              userActuallyDeleted: false,
-            });
-          }
-        });
-      }),
+      eventEmitter
+        .observeEvent('deleteProfileAndShowSuccessModal')
+        .subscribe(() => {
+          signOutAndDeleteAccount().then((success) => {
+            if (success) {
+              this.setState({
+                userDeletedModalOpened: true,
+                userActuallyDeleted: true,
+              });
+            } else {
+              this.setState({
+                userDeletedModalOpened: true,
+                userActuallyDeleted: false,
+              });
+            }
+          });
+        }),
 
       openSignUpInModal$.subscribe(({ eventValue: metaData }) => {
         // Sometimes we need to still open the sign up/in modal
         // after login is completed, if registration is not complete.
         // But in that case, componentDidUpdate is somehow called before
         // the modal is closed which overwrites the metaData.
         // This slightly dirty hack covers that case.
-        if (metaData) return;
-
-        setTimeout(() => {
-          this.forceUpdate();
-        }, 1);
+        if (metaData) {
+          return;
+        } else {
+          // if metaData is undefined, it means we're closing
+          // the sign up/in modal.
+          this.setState({ signUpInModalMounted: false });
+          setTimeout(() => {
+            this.forceUpdate();
```

Comment:
I'm not a fan of this but perhaps there's a good reason to do it this way? Why do we force an update in a timeout?

Reply:
Not a fan either, but I didn't touch this part to not cause unexpected bugs + increase the scope of this PR (which is why I'm not of fan). The comment above explains it, partly, I think? You could also ask Luuc in the front-end channel and take it further there.